core/trino-main/src/main/java/io/trino/security/AccessControl.java

```diff
@@ -222,10 +222,15 @@
     void checkCanShowColumns(SecurityContext context, CatalogSchemaTableName table);
 
     /**
-     * Filter the list of columns to those visible to the identity.
+     * Filter the list of columns to those visible to the identity when querying metadata.
      */
     Set<String> filterColumns(SecurityContext context, CatalogSchemaTableName tableName, Set<String> columns);
 
+    /**
+     * Filter the list of columns to those visible to the identity when querying the table.
+     */
+    Set<String> filterTableSchema(SecurityContext context, CatalogSchemaTableName tableName, Set<String> columns);
```

Comment:
Should this have a default implementation to simply return `columns`?

Reply:
I was wondering that too, but it seems that none of the methods in this interface have a default implementation. Since the implementation here is going to be pretty similar to `filterColumns` I just followed that. 

Do you think having defaults here would be better?

Also, I was curious why the default implementation for `filterColumns()` in `SystemAccessControl` and `ConnectorAccessControl` returned `emptySet()`. I would have expected it to return `columns` if anything. Is default behavior just to deny access where possible?

Reply:
Right, I was too quick I confused this class with `SystemAccessControl`. `SystemAccessControl` should have a default and here is ok.